src/commands/christmas.ts

```diff
@@ -1,90 +1,117 @@
+const fileFormatsToCheck = ['jpg', 'jpeg', 'png', 'gif']
+
+const getLastPinned = async () => {
+    try {
+        const lastPinned = await fs.readFile(lastPinnedId, 'utf8')
+        return lastPinned
+    } catch (e) {
+        return '0'
+    }
+}
+
+const setLastPinned = async (id: string | number) => {
+    await fs.writeFile(lastPinnedId, id.toString())
+}
 
 export default (commander: Commander): void => {
     commander.addCommand({
         commands: ['christmas', 'ch', 'luukku'],
         arguments: [],
         help: 'Avaa päivän joulukalenteriluukun',
 
-        func: (args, message, telegram) => {
+        func: async (args, message, telegram) => {
             const today = dayjs()
-            const christmasDay = dayjs('2020-12-25')
+            const christmas = dayjs().set('month', 11).set('date', 25)
 
-            if (today.get('month') !== 11) {
+            if (dayjs().get('month') !== 11) {
                 telegram.sendMessage(message.chat.id, 'Ei nyt ole joulukuu!')
-            } else if (today.format() === christmasDay.format()) {
-                telegram.sendMessage(message.chat.id, 'Hyvää joulua!', {
+                return
+            }
+
+            if (today.isAfter(christmas)) {
+                telegram.sendMessage(message.chat.id, 'Joulu meni jo :D', {
                     disable_notification: true,
                     parse_mode: 'html',
                 })
-            } else if (today.format() > christmasDay.format()) {
-                telegram.sendMessage(message.chat.id, 'Joulu meni jo :D', {
+                return
+            }
+
+            if (christmas.isToday()) {
+                telegram.sendMessage(message.chat.id, 'Hyvää joulua!', {
                     disable_notification: true,
                     parse_mode: 'html',
                 })
-            } else {
-                let message_id: number
-
-                telegram.sendMessage(message.chat.id, 'Avataan luukkua!!').then((res) => (message_id = res.message_id))
-
-                /*
-        // Sending local photos is currently not supported
-        telegram
-          .sendLocalPhoto(update.chat, images[today.get('date') - 1], `Luukku ${today.get('date')}`, {
-            disable_notification: true,
-            parse_mode: 'html',
-          })
-          .then(res => {
-            if (message_id) {
-              telegram.deleteMessage(update.chat.id, message_id)
+                return
             }
 
-            if (lastDate !== today.date()) {
-              telegram.pinChatMessage(update.chat.id, res.result.message_id, { disable_notification: true })
-            }
-            lastDate = today.date()
-          })
-          .catch(reason => {
-            console.log('Failed to send image')
-          })
-        */
+            const pre_res = await telegram.sendMessage(message.chat.id, 'Avataan luukkua!!')
+            const message_id = pre_res.message_id
+
+            try {
```

Comment:
Such a good code. This writer has to be a god. I really cant believe it

Reply:
Yeah sure